ScienceJournal/UI/ExperimentItemsViewController.swift

```diff
@@ -436,9 +436,7 @@ class ExperimentItemsViewController: VisibilityTrackingViewController,
   func collectionView(_ collectionView: UICollectionView,
                       layout collectionViewLayout: UICollectionViewLayout,
                       insetForSectionAt section: Int) -> UIEdgeInsets {
-    if experimentDataSource.isArchivedFlagSection(section) &&
-        experimentDataSource.numberOfItemsInSection(
-        experimentDataSource.archivedFlagSectionIndex) == 0 {
+    if experimentDataSource.numberOfItemsInSection(section) == 0 {
```

Comment:
So as discussed, the one issue with this is that, depending on the sections visible, you can still end up with a case where a section has a bottom padding and the next section has a top padding, which causes a double-padding scenario (such as having items in an experiment then starting a recording). Here's a proposed solution that _seems_ to solve this in all cases:

```
    // Determine the last section with items.
    var lastSectionWithItems:Int? = nil
    for i in 0..<collectionView.numberOfSections {
      if collectionView.numberOfItems(inSection: i) > 0 {
        lastSectionWithItems = i
      }
    }

    if experimentDataSource.numberOfItemsInSection(section) == 0 {
      // If the section is empty, no insets.
      return .zero
    } else if section == lastSectionWithItems {
      // If this is the last section, include the bottom inset.
      return cellInsets
    } else {
      // If this is not the bottom section, but this section has items, use standard insets without
      // bottom. This avoids an issue where two stacked sections
      var modifiedInsets = cellInsets
      modifiedInsets.bottom = 0
      return modifiedInsets
    }
```